hive3/src/main/java/org/apache/iceberg/mr/hive/serde/objectinspector/IcebergDateObjectInspectorHive3.java

```diff
@@ -69,4 +69,8 @@ public Object copyObject(Object o) {
     }
   }
 
+  @Override
+  public LocalDate convert(Object o) {
+    return o == null ? null : LocalDate.of(((Date) o).getYear(), ((Date) o).getMonth(), ((Date) o).getDay());
```

Comment:
nit: can we cast to Date only once and reuse that? might help slightly with readability too

Reply:
Fixed.